content/dvacaty_paty_sraz.md

```diff
@@ -0,0 +1,62 @@
+
+### Co musíme udělat, aby hra konečně fungovala jako hra?
+
+Karolina položila důležitou otázku. Co je třeba udělat, aby hra byla hratelná.  Tedy jaké úkoly máme pro příští týdny. 
+Hlavní je nyní kompletně hru předělat do podoby server - klient. V současné chvíli máme jednoduchý prototyp serveru, který po připojení klienta umí poslat soubor. 
+Máme jednoduchého klienta, který se umí k serveru připojit a přijmout soubor. Jedná se zatím o jednoduchý soubor ve formátu JSON. Tento typ klienta, který dostává popis stavu hry, by jej měl dokázat převést do grafické podoby. Tedy zobrazit aktuální stav hry na základě dat dodaných serverem. Dnes jsme se tedy zaměřily na tento úkol. Soubor, který nyní server odesílá, obsahuje dohromady informace o herním plánu a robotech. Pro vykreslení je potřebujeme rozdělit a získat tak podklady pro vytvoření `board` a `robots`. 
+Pro vytvoření `board` již máme funkci napsanou. Abychom ji ale mohly využít i v modelu server - klient, je třeba ji trošku upravit, aby odpovídala vstupním datům, která bude posílat server. 
+Dostaly jsme se zde do situace, kdy tedy potřebujeme jednu z původních funkci rozdělit na dvě, abychom mohly použít jen její část, ale nechceme tím rozbít existující testy a zbytek kódu. Petr nám tedy poradil jednoduchý způsob.
+Najdeme si místo, kde chceme funkci rozdělit, a určíme tak, kterou část má dělat nová funkce. Naše nová funkce pak dostane jako parametr vše, co dělá první část kódu. A v první části kódu pak novou funkci zavoláme. Přidáme tedy v podstatě jen dva řádky kódu (viz ilustrační obrázek) a původní funkce tedy dělá to, co předtím, nová funkce se ale dá bez problémů použít samostatně zvlášť. 
+
+![rozdeleni](./images/rozdeleni_fce.jpg)
+
+### Logika server - klient
+
+Potřebovaly jsme si trošku ujasnit, jak přesně bude fungovat, když hru přepíšeme tak, aby běžela na serveru a hráči se k ní připojovali přes klienty.
+Server zpracovává samotnou logiku hry a drží všechny informace o ní, posílá je přihlášeným klientům a také zpracovává informace, které od nich dostane. Klienty budeme mít dvou typů. První bude čistě zobrazovací, prozatím mu říkáme `receiver`. Jeho úkolem bude pouze vykreslovat aktuální stav hry, tedy herní plochu a roboty. Druhý, o něco složitější klient, bude pro ty, kteří chtějí hru hrát, a poběží na něm náš současný interface. Říkáme mu `sender`, protože prostřednictvím tohoto klienta budou hráči dostávat od serveru karty, programovat roboty a odesílat data zpět na server. Zobrazení herní plochy a robotího interface budou tedy dva nezávislé programy. 
+
+Celý projekt je nakonec výrazně komplexnější a náročnější, než jsme si na začátku dovedly představit. Ale výsledek bude určitě stát za to! :)
+
+![owl](./images/owl.jpg)
```

Comment:
Jak je to s copyrightem, dá se tenhle obrázek šířit?

Reply:
Aha! Zjistím

Reply:
Nemohu to najít s jasně definovanou licencí nebo pravidly šíření, tak to celé dám pryč, není to tam stejně k ničemu dobré, jen jsme tím tak hezky končili tu lekci :)